src/api/auth/src/user.js

```diff
@@ -1,23 +1,98 @@
-// Regular (non-admin) User
+// A User represents a Seneca SSO Authenticated user who might also be a Telescope user.
 class User {
-  constructor(name, email, id, nameID, nameIDFormat) {
-    this.name = name;
-    this.email = email;
-    this.id = id;
-    this.nameID = nameID;
-    this.nameIDFormat = nameIDFormat;
-    this.isAdmin = false;
+  constructor(senecaProfile, telescopeProfile) {
+    // TODO: what to do about id?  Hash it like before or use email? hash(profile.nameID)
+    // this.id = id;
+
+    // All authenticated users are Seneca Users
+    this.seneca = {
+      'http://schemas.microsoft.com/identity/claims/displayname':
+        senecaProfile['http://schemas.microsoft.com/identity/claims/displayname'],
+      'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress':
+        senecaProfile['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'],
+      nameID: senecaProfile.nameID,
+      nameIDFormat: senecaProfile.nameIDFormat,
+    };
+
+    // But not all users are Telescope Users (yet) if they haven't signed up.
+    if (telescopeProfile) {
+      this.telescope = telescopeProfile;
+    }
+  }
+
+  // Normalize access to Standard Seneca user properties and optional Telescope properties
+  get email() {
+    return this.seneca['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'];
+  }
+
+  get nameID() {
+    return this.seneca.nameID;
+  }
+
+  get nameIDFormat() {
+    return this.seneca.nameIDFormat;
+  }
+
+  get firstName() {
+    return this.telescope?.firstName;
+  }
+
+  get lastName() {
+    return this.telescope?.lastName;
+  }
+
+  get isAdmin() {
+    return !!(this.telescope && this.telescope.isAdmin === true);
   }
 
+  get isFlagged() {
+    return !!(this.telescope && this.telescope.isFlagged === true);
+  }
+
+  // Prefer Telescope Display Name if set, but fallback to Seneca's
+  get displayName() {
+    return (
+      this.telescope?.displayName ||
+      this.seneca['http://schemas.microsoft.com/identity/claims/displayname']
+    );
+  }
+
+  get feeds() {
+    return this.telescope?.feeds;
+  }
+
+  get github() {
+    return this.telescope?.github;
+  }
+
+  get isTelescopeUser() {
+    return !!this.telescope;
+  }
+
+  // Get a list of roles for this user
+  get roles() {
```

Comment:
In https://github.com/Seneca-CDOT/telescope/pull/2058/files#diff-2c521e1745fe5dd3ad9df3a78ef80c8cd45f49c62e8367ab03026a5d7549df92R22 you say we have 4 roles, but this function only returns up to 3. Is there any chance we'd need the service role returned as well?

Reply:
No, we'd never create a `service` token from the user authentication flow.  See https://github.com/Seneca-CDOT/satellite/pull/10, and add a review, if you're willing.